### ISSUE TITLE
raw_attribute and raw_tex no longer supported for org reader?

### ISSUE THREAD

**Issue by nfleisher** [jgm/pandoc#5928]
Just upgraded to pandoc 2.8 and it seems that the org reader no longer supports the `raw_attribute` extension. I had been using that extension to include latex code in the `header-includes` field of a yaml metadata file (mainly `\usepackage` directives). But now that usage is broken.

Is there another way to put these pieces together (org source file, yaml metadata file, latex output) in such a way that one can put latex code in `header-includes`? I can always move that stuff into the template, but `header-includes` is far more versatile and it was nice functionality to have available via `raw_attribute`. Would you consider adding `raw_attribute` support back to the org reader (assuming it doesn't conflict with other changes)?

(Update: I tried using the `raw_tex` extension instead, and pandoc 2.8 throws an error saying that `raw_tex` is not supported for org, just as it does for `raw_attribute`. Org is explicitly listed as a supported input format for the `raw_tex` extension in the pandoc 2.8 man page.)

**Comment by jgm** [comment]
Ah! I see what is happening.  `raw_attribute` is indeed not supported in org, and never has been: it's a markdown extension.  But you're using a markdown-formatted YAML metadata file via `--metadata-file`, I gather?

There were two relevant changes in 2.8 that cause this:

1. The YAML metadata is parsed using `readerOpts` -- previously we just had a fixed set of options for parsing metadata files, and you couldn't adjust it.
2. We now raise an error for unuspported extensions.

Because `raw_attribute` isn't supported for `org`, you can't add it (2), but this means it isn't used in parsing the metadata file (1).  Obviously this isn't good for cases where the main format isn't markdown.  I hadn't considered this.

See #5279 and the issues linked there for background.
@mb21 do you have thoughts about how to deal with this?

**Comment by jgm** [comment]
For now you could try using the new `--defaults` feature.  Create a file whatever.yaml
with
```
include-in-header: |
  the raw stuff you want to include
```
and use with `-d whatever`.

**Comment by nfleisher** [comment]
> Ah! I see what is happening. `raw_attribute` is indeed not supported in org, and never has been: it's a markdown extension. But you're using a markdown-formatted YAML metadata file via `--metadata-file`, I gather?

Yes, exactly. Here's an example of what I had been successfully including in that YAML metadata file up until 2.8, in case it's helpful. Thanks for the suggestion about `--defaults`.

```
header-includes:
- |
  `\usepackage{nf-linguistics}
  \usepackage{nf-semantics}
  \usepackage{nf-syntax}
  \usepackage[italic]{mathastext}`{=latex}
```

**Comment by jgm** [comment]
This interacts with #5914.
If we paid attention to a `format_` metadata field in metadata files, users could explicitly specify the format and extensions they intend.

**Comment by nfleisher** [comment]
> This interacts with #5914.
> If we paid attention to a `format_` metadata field in metadata files, users could explicitly specify the format and extensions they intend.

That kind of flexibility would be nice to have. For the workflow I was describing above---converting from org to latex and including some specific latex options via `header-includes`---the sensible default format would be the output format, not the input format. But I imagine there are other workflows and/or other values one might set in the YAML metadata file where the input format would be the sensible default.

**Comment by jgm** [comment]
> the sensible default format would be the output format, not the input format

That's only because of the header-includes.  But these aren't really metadata, conceptually.  So I think moving to use of a `--default` file might make more sense here.

**Comment by tarleb** [comment]
I think this can be closed in favor of #5914, which would solve this.

For LaTeX headers one could also use the `#+LATEX_HEADER` org directive. See also <https://pandoc.org/org.html>.
